rediscluster.rb

```diff
@@ -191,7 +191,19 @@ def send_cluster_command(argv, master_only: true, &blk)
         r = @connections.get_connection_by_node(ask_redirect_node)
         ask_redirect_node = nil
       else
-        r = @connections.get_connection_by_slot(slot, master_only)
+        # In a rare situation where a shard is failing over
+        # there is a chance that some slots are unassigned.
+        # In this situation, we should try connecting to a random node.
+        # That will either trigger a "MOVED" or "ASK" response
+        # and hopefully we will reach the appropriate node within the limited iterations
+        r =
+          if @connections.slot_has_nodes?(slot)
+            @connections.get_connection_by_slot(slot, master_only)
+          else
+            puts "Cannot find nodes for slot #{slot}, trying with a random node"
+            sleep(@cluster_retry_interval) if ttl < RedisClusterRequestTTL/2
```

Comment:
I have kept the sleep conditional similar to the pattern adopted in rest of the file

Reply:
you have puts on it

Reply:
removed now